src/stirling/source_connectors/socket_tracer/protocols/mqtt/stitcher_test.cc

```diff
@@ -0,0 +1,623 @@
+
+  RecordsWithErrorCount<Record> result;
+
+  Message connect_frame, connack_frame, pingresp_frame;
+  connect_frame = CreateFrame(kRequest, 1, 0, 0);
+  connack_frame = CreateFrame(kResponse, 2, 0, 0);
+  pingresp_frame = CreateFrame(kResponse, 13, 0, 0);
+
+  int t = 0;
+  connect_frame.timestamp_ns = ++t;
+  connack_frame.timestamp_ns = ++t;
+  pingresp_frame.timestamp_ns = ++t;
+  req_map[0].push_back(connect_frame);
+  resp_map[0].push_back(connack_frame);
+  resp_map[0].push_back(pingresp_frame);
+  result = StitchFrames(&req_map, &resp_map);
+  EXPECT_TRUE(AreAllDequesEmpty(resp_map));
+  EXPECT_EQ(TotalDequeSize(req_map), 0);
+  EXPECT_EQ(result.error_count, 1);
+  EXPECT_EQ(result.records.size(), 1);
+}
+
+TEST(MqttStitcherTest, InOrderMatching) {
+  // Test for packet stitching for packets that do not have packet identifiers
+  absl::flat_hash_map<packet_id_t, std::deque<Message>> req_map;
+  absl::flat_hash_map<packet_id_t, std::deque<Message>> resp_map;
+
+  RecordsWithErrorCount<Record> result;
+
+  Message connect_frame, connack_frame, pingreq_frame, pingresp_frame, pub1_frame, pub1_pid3_frame,
+      puback_frame, puback_pid3_frame, pub2_frame, pub2_pid2_frame, pubrec_frame, pubrec_pid2_frame,
+      pubrel_frame, pubrel_pid2_frame, pubcomp_frame, pubcomp_pid2_frame, sub_frame, suback_frame,
+      unsub_frame, unsuback_frame;
+  connect_frame = CreateFrame(kRequest, 1, 0, 0);
+  connack_frame = CreateFrame(kResponse, 2, 0, 0);
+  pingreq_frame = CreateFrame(kRequest, 12, 0, 0);
+  pingresp_frame = CreateFrame(kResponse, 13, 0, 0);
+  pub1_frame = CreateFrame(kRequest, 3, 1, 1);
+  pub1_pid3_frame = CreateFrame(kRequest, 3, 3, 1);
+  pub2_frame = CreateFrame(kRequest, 3, 1, 2);
+  pub2_pid2_frame = CreateFrame(kRequest, 3, 2, 2);
+  puback_frame = CreateFrame(kResponse, 4, 1, 0);
+  puback_pid3_frame = CreateFrame(kResponse, 4, 3, 0);
+  pubrec_frame = CreateFrame(kResponse, 5, 0, 0);
+  pubrec_pid2_frame = CreateFrame(kResponse, 5, 2, 0);
+  pubrel_frame = CreateFrame(kRequest, 6, 0, 0);
+  pubrel_pid2_frame = CreateFrame(kRequest, 6, 2, 0);
+  pubcomp_frame = CreateFrame(kResponse, 7, 0, 0);
+  pubcomp_pid2_frame = CreateFrame(kResponse, 7, 2, 0);
+  sub_frame = CreateFrame(kRequest, 8, 0, 0);
+  suback_frame = CreateFrame(kResponse, 9, 0, 0);
+  unsub_frame = CreateFrame(kRequest, 10, 0, 0);
+  unsuback_frame = CreateFrame(kResponse, 11, 0, 0);
+
+  int t = 0;
+  connect_frame.timestamp_ns = ++t;
+  connack_frame.timestamp_ns = ++t;
+  pingreq_frame.timestamp_ns = ++t;
+  pingresp_frame.timestamp_ns = ++t;
+  req_map[0].push_back(connect_frame);
+  resp_map[0].push_back(connack_frame);
+  req_map[0].push_back(pingreq_frame);
+  resp_map[0].push_back(pingresp_frame);
+  result = StitchFrames(&req_map, &resp_map);
+  EXPECT_TRUE(AreAllDequesEmpty(resp_map));
+  EXPECT_EQ(TotalDequeSize(req_map), 0);
+  EXPECT_EQ(result.error_count, 0);
+  EXPECT_EQ(result.records.size(), 2);
+  req_map.clear();
+  resp_map.clear();
```

Comment:
Previous assertions prove that the maps are cleared, please remove.

Reply:
Yeah sure! I will make sure to remove redundant test cases.